Sources/NIOCore/EventLoopFuture+AssumeIsolated.swift

```diff
@@ -119,26 +130,54 @@ struct IsolatedEventLoop {
 
     /// Returns the wrapped event loop.
     @inlinable
-    func nonisolated() -> any EventLoop {
+    public func nonisolated() -> any EventLoop {
```

Comment:
Can we make this method also `noasync`

Reply:
I don't think we need to: this is the only method that is _always_ safe to call on this type.

Reply:
I understand your argument my reason is that if a user ever ends up holding this type in an `async` method I really want them to see an error. Even if that place is not where the error happened but they should question how they got the type there in the first place.